imap_processing/idex/IDEXPacketParser.py

```diff
@@ -0,0 +1,120 @@
+import bitstring
+import numpy as np
+from space_packet_parser import parser, xtcedef
+from imap_processing import packet_definition_directory
+
+
+class IDEXPacketParser():
+    def __init__(self, idex_packet_file: str):
+
+        idex_xtce = f"{packet_definition_directory}/idex_packet_definition.xml"
+        idex_definition = xtcedef.XtcePacketDefinition(xtce_document=idex_xtce)
+        idex_parser = parser.PacketParser(idex_definition)
+
+        idex_binary_data = bitstring.ConstBitStream(filename=idex_packet_file)
+        idex_packet_generator = idex_parser.generator(idex_binary_data)
+
+        self.epochs = []
+        self.data = {}
+        
+        evtnum = 0
+        for pkt in idex_packet_generator:
+            if 'IDX__SCI0TYPE' in pkt.data:
+                scitype = pkt.data['IDX__SCI0TYPE'].raw_value
+                if scitype == 1:
+                    evtnum += 1
+                    self.epochs.append(pkt.data['SHCOARSE'].derived_value + 20*(10**(-6))*pkt.data['SHFINE'].derived_value)  # Use this as the CDF epoch
```

Comment:
It might be good to add a comment or assign those numbers (20, 10, -6) to variables, so it's clear where they're coming from.